src/storage/mod.rs

```diff
@@ -425,7 +425,15 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
         mut ctx: Context,
         raw_key: Vec<u8>,
         start_ts: TimeStamp,
-    ) -> impl Future<Output = Result<(Option<Value>, Statistics, PerfStatisticsDelta)>> {
+    ) -> impl Future<
+        Output = Result<(
+            Option<Value>,
+            Statistics,
+            PerfStatisticsDelta,
+            KvGetLatencyStats,
+        )>,
+    > {
+        let stage_begin_ts = Instant::now();
```

Comment:
`now_coarse` instead?

Reply:
at L436 start schedule waiting

at L447 finish schedule waiting
```rust
        let res = self.read_pool.spawn_handle(
            async move {
                // finished schedule wait here.

                tls_collect_query(
                    ctx.get_region_id(),
                    ctx.get_peer(),
                    key.as_encoded(),
                    key.as_encoded(),
                    false,
                    QueryKind::Get,
                );
            }

            // ...
        );
```

Reply:
> 

I learned that `wait_wall_time = schedule_wait_time + snapshot_wait_time`, and we seem to have no need to return `schedule_wait_time` , so I did not get unnecessary timestamp at the stage of `Scheduled`.

Reply:
`now_coarse()` seems to have less precision, is it acceptable here?

Reply:
> 

We can split them first and merge when responding.

Reply:
We get `wait_wall_time` by sub `begin_ts` from `snapshot_received_ts`, it seems unnecessary to calculate by adding `schedule_wait_time` and `snapshot_wait_time`. But I can put a temporarily useless `_stage_scheduled_ts` here for future needs.

Reply:
> `now_coarse()` seems to have less precision, is it acceptable here?

Get request is more latency-sensitive than coprocessor. As you can see, most timing methods for coprocessor are `now_coarse`s, so it makes sense to get requests to act the same.

Reply:
> Get request is more latency-sensitive than coprocessor. As you can see, most timing methods for coprocessor are now_coarses, so it makes sense to get requests to act the same.

Got it

Reply:
> We get `wait_wall_time` by sub `begin_ts` from `snapshot_received_ts`, it seems unnecessary to calculate by adding `schedule_wait_time` and `snapshot_wait_time`. But I can put a temporarily useless `_stage_scheduled_ts` here for future needs.

I prefer to place it to `KvGetLatencyStats`.

Reply:
done
[src/storage/mod.rs#L440](https://github.com/tikv/tikv/pull/11508/files#diff-e3216d051c82e0ef881cdaeee1b11ef716ff9d72655f381da72a7828e9ca4b3cR440)
[src/storage/mod.rs#L530](https://github.com/tikv/tikv/pull/11508/files#diff-e3216d051c82e0ef881cdaeee1b11ef716ff9d72655f381da72a7828e9ca4b3cR530)